Add docs

lilyminium opened the pull request:
Fixes #18

This PR:

* fleshes out the docs a bit, and removes info from the README
* automatically generates the terminal log and the repository tree as part of docs creation
* Moves docs into a subfolder called `source` to keep things tidier
* Adds RTD config
* Adds doc building and deployment to CI -- to gh pages
* Adds some of the niceties in the cookie -- CHANGELOG, AUTHORS, etc. I retrofilled the CHANGELOG
* adds a couple comments to CI

pep8speaks commented:
Hello @lilyminium! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`docs/conf.py`](https://github.com/MDAnalysis/cookiecutter-mdakit/blob/f88de235c44d420661535854c7a0b67a118946fa/docs/conf.py):

> [Line 22:1](https://github.com/MDAnalysis/cookiecutter-mdakit/blob/f88de235c44d420661535854c7a0b67a118946fa/docs/conf.py#L22): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file

lilyminium commented:
@IAlibay @orbeckst @fiona-naughton I added some docs, deployment, RTD, comments, etc. Would appreciate any feedback!

Again this is quite a large PR but a lot of the LOC is from scripts used to auto-generate three pieces of documentation:

* the terminal output here: https://cookiecutter-mdakit--30.org.readthedocs.build/en/30/usage.html#use-the-cookiecutter-to-create-a-new-project
* the tree right below it
* the author list in the bottom has been turned into a script.

So the best way to look at this PR might be to look at the RTD build and just see if you're happy with it. I replaced most of the original MolSSI docs.

I'm not quite sure what else to put in the documentation, really. Perhaps an explanation of the cookie CI.
